azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -92,9 +95,11 @@
 
    private final AzureComputeConstants azureComputeConstants;
 
+   private final ProviderMetadata providerMetadata;
+
    @Inject
    AzureComputeServiceAdapter(final AzureComputeApi api, final AzureComputeConstants azureComputeConstants,
-                              CleanupResources cleanupResources) {
+                              CleanupResources cleanupResources, ProviderMetadata providerMetadata) {
```

Comment:
Better inject directly the `@Nullable @Named(LocationConstants.PROPERTY_ZONES) String enabledZones` and split the property and assign the list in the constructor.